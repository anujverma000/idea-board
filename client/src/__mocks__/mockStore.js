import configureStore from 'redux-mock-store'
import thunk from "redux-thunk";
import rootReducer from "../reducers";

const initialState= {
    idea: {
        list: []
    }
}
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);
store.replaceReducer(rootReducer);

export default store;