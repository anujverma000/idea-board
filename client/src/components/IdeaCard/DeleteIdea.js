import React from "react";
import { connect } from 'react-redux';
import styled from "styled-components";
import { ReactComponent as TrashIcon } from './trash.svg';
import { deleteIdea } from '../../actions/ideaActions';

const DeleteIdea = ({ idea, trashIdea }) => {

  const handleDelete = e => trashIdea(idea.id);
  
  return(
    <Wrapper onClick={handleDelete}>
      <TrashIcon/>
    </Wrapper>
  )
}


/**
 *  STYLES
 */
const Wrapper = styled.span`
  position: absolute;
  top: 8px;
  right: 8px;
`;

const mapDispatchToProps = dispatch => ({
  trashIdea: id =>  dispatch(deleteIdea(id))
});


export default connect(null, mapDispatchToProps)(DeleteIdea);