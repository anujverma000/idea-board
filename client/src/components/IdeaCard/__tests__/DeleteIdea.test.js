import React from 'react';
import renderer from 'react-test-renderer';
import DeleteIdea from '../DeleteIdea'
import mockStore from "../../../__mocks__/mockStore";
import { Provider } from "react-redux";

it('renders correctly', () => {
  const tree = renderer.create(
      <Provider store={mockStore}><DeleteIdea /></Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
