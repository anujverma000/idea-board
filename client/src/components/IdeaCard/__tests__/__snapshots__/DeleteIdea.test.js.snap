// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="sc-bdVaJa fnZpRG"
  onClick={[Function]}
>
  <svg>
    trash.svg
  </svg>
</span>
`;
